instrumentation/netty/netty-4.1/library/src/main/java/io/opentelemetry/instrumentation/netty/v4_1/internal/client/HttpClientResponseTracingHandler.java

```diff
@@ -31,25 +46,24 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
     }
 
     Attribute<Context> parentContextAttr = ctx.channel().attr(AttributeKeys.CLIENT_PARENT_CONTEXT);
-    Attribute<HttpRequestAndChannel> requestAttr =
-        ctx.channel().attr(NettyClientSingletons.HTTP_REQUEST);
+    Attribute<HttpRequestAndChannel> requestAttr = ctx.channel().attr(HTTP_REQUEST);
 
     Context parentContext = parentContextAttr.get();
     HttpRequestAndChannel request = requestAttr.get();
 
     if (msg instanceof FullHttpResponse) {
-      parentContextAttr.remove();
-      contextAttr.remove();
-      requestAttr.remove();
+      parentContextAttr.set(null);
+      contextAttr.set(null);
+      requestAttr.set(null);
```

Comment:
for perf? nice

Reply:
These methods were deprecated, and it appears that the library conventions are more strict about not using deprecated methods than the javaagent convention because the build fails without this.